### ISSUE TITLE
请问yolov4原论文提到的方法都在这里实现了吗？

### ISSUE THREAD

**Issue by TMRin** [bubbliiiing/yolov4-pytorch#48]
下面是yolov4原论文中给出的方法，请问您的代码全部实现了这些功能吗？如果没有的话具体哪些还没有实现呢？谢谢！

YOLO v4 uses:
• Bag of Freebies (BoF) for backbone: CutMix and Mosaic data augmentation, DropBlock regularization,Class label smoothing
• Bag of Specials (BoS) for backbone: Mish activation, Cross-stage partial connections (CSP), Multi input weighted residual 
  connections (MiWRC)
• Bag of Freebies (BoF) for detector: CIoU-loss,CmBN, DropBlock regularization, Mosaic data augmentation, Self-Adversarial 
  Training, Eliminate grid sensitivity, Using multiple anchors for a single ground truth, Cosine annealing scheduler [52], Optimal 
  hyperparameters, Random training shapes
• Bag of Specials (BoS) for detector: Mish activation,SPP-block, SAM-block, PAN path-aggregation block,DIoU-NMS

**Comment by bubbliiiing** [comment]
1、主干特征提取网络：DarkNet53 => CSPDarkNet53

2、特征金字塔：SPP，PAN

3、分类回归层：YOLOv3（未改变）

4、训练用到的小技巧：Mosaic数据增强、Label Smoothing平滑、CIOU、学习率余弦退火衰减

5、激活函数：使用Mish激活函数

**Comment by bubbliiiing** [comment]
并没有实现全部，很多非常隐晦，不容易理解

**Comment by TMRin** [comment]
> 并没有实现全部，很多非常隐晦，不容易理解
好的，非常感谢！

**Comment by TMRin** [comment]
> 并没有实现全部，很多非常隐晦，不容易理解

好的，非常感谢！
